haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -829,6 +855,43 @@ def _split_sentences(self, text: str) -> List[str]:
         sentences = sentence_tokenizer.tokenize(text)
         return sentences
 
+    def _split_tokens(self, text: str, tokenizer: Any) -> List[str]:
+        if tokenizer == "tiktoken":
+            try:
+                import tiktoken
+            except ImportError:
+                raise ImportError(
+                    "Tiktoken is needed in order to split documents by tokens. "
+                    "Please install it with `pip install tiktoken`."
+                )
+            enc = tiktoken.get_encoding("cl100k_base")
+            integer_tokens = enc.encode(text, allowed_special="all", disallowed_special=())
+            elements = [enc.decode_single_token_bytes(token).decode() for token in integer_tokens]
+        elif isinstance(tokenizer, str):
+            try:
+                from transformers import AutoTokenizer
+            except ImportError:
+                raise ImportError(
+                    "HuggingFace transformers is needed in order to split documents using HuggingFace tokenizers. "
+                    "Please install it with `pip install transformers`."
+                )
+            try:
+                tokenizer = AutoTokenizer.from_pretrained(tokenizer)
+            except Exception:
+                raise ValueError(
+                    f"Could not load tokenizer '{tokenizer}' from HuggingFace model hub. "
+                    f"Please make sure that the tokenizer is correct and exists."
+                )
+            elements = tokenizer.tokenize(text)
+        elif isinstance(tokenizer, PreTrainedTokenizerBase):
+            elements = tokenizer.tokenize(text)
+        else:
+            raise ValueError(
+                f"Unsupported tokenizer specification {tokenizer}. "
+                f"Please provide either the string 'tiktoken' or a HuggingFace tokenizer (PreTrainedTokenizerBase)."
+            )
+        return elements
```

Comment:
Nitpick: if you return elements into each if block, you can avoid the `elif`s and the `else` at the end. Let's simplify this block.

Reply:
Done.